### PR TITLE
fix: Avoid cozy-ui stylesheet conflict with the apps

### PR DESCRIPTION
**Pull request by drazik**
This PR goal is to fix the bugs we have when we use a cozy-bar with a cozy-ui version that is not the same as the host app cozy-ui version.

This problems exists because the cozy-bar uses the transpiled version of cozy-ui, thus we could have two stylesheets that could override each other.

The solution explored here is to stop using the transpiled version of cozy-ui. The bar transpiles its own cozy-ui and use a custom classname scheme for cozy-ui's classnames. This way, there is no more conflicts between the bar's cozy-ui stylesheets and app's cozy-ui stylesheet.

This was tested on Banks, in which we had this issue: they are all gone.

### CODE REVIEW

File: config/webpack.config.base.js [hunk 1]
```diff
@@ -14,20 +14,22 @@ module.exports = {
     umdNamedDefine: true
   },
   resolve: {
-    extensions: ['.js', '.json', '.yaml'],
+    extensions: ['.js', '.json', '.yaml', '.jsx'],
     modules: [SRC_DIR, 'node_modules'],
     alias: {
       react: path.resolve(__dirname, 'aliases/globalReact'),
-      'react-dom': path.resolve(__dirname, 'aliases/globalReactDOM'),
-      'cozy-ui/react': 'cozy-ui/transpiled/react'
```
Comment (ptbrowne) [inline config/webpack.config.base.js]: If I am not mistaken, the most important part of this commit is the removal of the alias.

Nitpick: The name of the commit is a bit confusing, we are not "transpiling", we are bundling "cozy-ui" through its non transpiled version :)

Comment (drazik) [reply config/webpack.config.base.js]: I splitted the commit into two.

**Comment by Crash--** [comment]
Related issues : 
#472 
#436

Seems a solid way to fix the issue. Let me build drive with it as we previously had some issues with the non transpiled version. 

Also, even if it's an internal changes we should considered this modification as at least a major version of the lib or as a breaking change.

**Comment by drazik** [comment]
> Also, even if it's an internal changes we should considered this modification as at least a major version of the lib or as a breaking change.

I think even if I did small chore commits, we should squash it as a fix with breaking changes when merging it.

**Comment by Crash--** [comment]
Seems to be ok for Drive/Android 👍

**Comment by drazik** [comment]
> Do you know if the css from the cozy-bar contains any global classes from cozy-ui? Presumably it still has some utility classes?

Yes, the cozy-bar stylesheet contains utility classes that are used in cozy-ui components that are used in the cozy-bar.

**Comment by y-lohse** [comment]
> Yes, the cozy-bar stylesheet contains utility classes that are used in cozy-ui components that are used in the cozy-bar.

Ok cool :) they shouldn't be a problem, but possibly the next issue to address.

File: config/webpack.config.base.js [hunk 2]
```diff
@@ -25,8 +25,11 @@ module.exports = {
   module: {
     rules: [
       {
-        test: /\.js$/,
-        exclude: /node_modules/,
+        test: /\.jsx?$/,
+        include: [
+          path.resolve(__dirname, '../src'),
```
Comment (gregorylegarec) [inline config/webpack.config.base.js]: I think we must also add our overrides directory here. Probably not in this file however.

**Pull request merged**